internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -1118,6 +1118,9 @@ func newAssistedCM(ctx context.Context, log logrus.FieldLogger, asc ASC) (client
 			"JWKS_URL":                    "https://api.openshift.com/.well-known/jwks.json",
 			"PUBLIC_CONTAINER_REGISTRIES": unauthenticatedRegistries(ctx, asc),
 			"HW_VALIDATOR_REQUIREMENTS":   defaultControllerHardwareRequirements,
+			// The user may opt-out by using the telemetry opt-out method (removing cloud.openshift.com from their OCM pull secret)
+			"ENABLE_DATA_COLLECTION": "True",
+			"DATA_UPLOAD_ENDPOINT":   "",
```

Comment:
Where does this actually get set?

As it is in this PR the user would have to provide this value in a configmap for this to ever work, right?

My preference would be for us to put the "real" value in our application as the default value and remove this line. We can always still override using the configmap.